### PR TITLE
MAGEDOC-3050: Composer install instructions

### PR DESCRIPTION
**Pull request by jeff-matthews**
## This PR is a:

- [ ] New topic
- [x] Content update
- [x] Content fix or rewrite
- [ ] Bug fix or improvement

## Summary

When this pull request is merged, it will consolidate four topics into one and provide a simplified set of instructions for using Composer to install Magento.

## Additional information

Here's what I did:

- [x] Created a new file `install-gde/composer.md`
- [x] Consolidated all four existing files into the new file
- [x] Rewrote the consolidated info
- [x] Deleted the old files
- [x] Replaced links to the old, deleted files with the new file link
- [x] Created redirects for the old, deleted files
- [x] Made these changes to 2.1 and later only; 2.0 remains unchanged
- [x] Created an internal CICD build for review (ping me and I'll send you a link or just use the branch name to infer the location)

### CODE REVIEW

File: guides/v2.1/install-gde/composer.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+-  Web Setup Wizard
+
+### Command line
+{:.no_toc}
+
+This example assumes that the Magento install directory is named `magento2ee`, the `db-host` is on the same machine (`localhost`), and that the `db-name`, `db-user`, and `db-password` are all `magento`:
+
+```bash
+bin/magento setup:install --base-url=http://localhost/magento2ee --db-host=localhost --db-name=magento --db-user=magento --db-password=magento --backend-frontname=admin --admin-firstname=admin --admin-lastname=admin --admin-email=admin@admin.com --admin-user=admin --admin-password=admin123 --language=en_US --currency=USD --timezone=America/Chicago --use-rewrites=1
+```
```
Comment (lorikrell) [inline guides/v2.1/install-gde/composer.md]: Format feedback: Not sure if possible, but would be great for this command to wrap on the generated page and still be copied and used. Readers can use a horizontal scrolled.

File: _includes/install/file-system-perms-oneuser.md [hunk 2]
```diff
@@ -22,7 +22,7 @@ To set permissions before you install the Magento software:
 3.	If you haven't done so already, get the Magento software in one of the following ways:
 
 	*	[Compressed archive]({{ page.baseurl }}/install-gde/prereq/zip_install.html)
-	*	[Composer metapackage]({{ page.baseurl }}/install-gde/prereq/integrator_install.html)
+	*	[Composer metapackage]({{ page.baseurl }}/install-gde/composer.html)
```
Comment (lorikrell) [inline _includes/install/file-system-perms-oneuser.md]: This include is also heavily used in 2.0. Might need to add conditional for version, build another include, other option?

File: _includes/install/composer-overview.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+-   Reuse third-party libraries without bundling them with source code
+-   Reduce extension conflicts and compatibility issues by using a component-based architecture with robust dependency management
+-   Adhere to [PHP-Framework Interoperability Group (FIG)](https://www.php-fig.org/){:target="\_blank"} standards
+-   Repackage Magento Open Source with other components
+-   Use the Magento software in a production environment
+
+{:.bs-callout .bs-callout-warning}
+You must create a Composer project from our metapackage if you want to use the Magento Web Setup Wizard to upgrade the Magento software and third-party extensions.
+</div>
```
Comment (lorikrell) [inline _includes/install/composer-overview.md]: I like how clean this include is now! Also verified it is used in 2.0, but the changes don't affect that content. Can use this same include for all.

File: _includes/install/file-system-perms-oneuser.md (hunk 2 above)
Comment (jeff-matthews) [reply _includes/install/file-system-perms-oneuser.md]: I don't think the changes I made require special handling for 2.0. I think they apply well to all versions.

File: _includes/install/file-system-perms-oneuser_22.md [hunk 4]
```diff
@@ -23,7 +23,7 @@ To set permissions before you install the Magento software:
 3.	If you haven't done so already, get the Magento software in one of the following ways:
 
 	*	[Compressed archive]({{ page.baseurl }}/install-gde/prereq/zip_install.html)
-	*	[Composer metapackage]({{ page.baseurl }}/install-gde/prereq/integrator_install.html)
+	*	[Composer metapackage]({{ page.baseurl }}/install-gde/composer.html)
```
Comment (lorikrell) [inline _includes/install/file-system-perms-oneuser_22.md]: Possible 2.0 using include? If used will 404 when clicked in 2.0 pages.

File: _includes/install/file-system-perms-twouser.md [hunk 5]
```diff
@@ -87,7 +87,7 @@ To complete the task, restart the web server:
 If you haven't done so already, get the Magento software in one of the following ways:
 
 *	[Compressed archive]({{ page.baseurl }}/install-gde/prereq/zip_install.html)
-*	[Composer metapackage]({{ page.baseurl }}/install-gde/prereq/integrator_install.html)
+*	[Composer metapackage]({{ page.baseurl }}/install-gde/composer.html)
```
Comment (lorikrell) [inline _includes/install/file-system-perms-twouser.md]: Possible 2.0 using include? If used will 404 when clicked in 2.0 pages.

File: _includes/install/file-system-perms-twouser_22.md [hunk 6]
```diff
@@ -87,7 +87,7 @@ To complete the task, restart the web server:
 If you haven't done so already, get the Magento software in one of the following ways:
 
 *	[Compressed archive]({{ page.baseurl }}/install-gde/prereq/zip_install.html)
-*	[Composer metapackage]({{ page.baseurl }}/install-gde/prereq/integrator_install.html)
```
Comment (lorikrell) [inline _includes/install/file-system-perms-twouser_22.md]: Possible 2.0 using include? If used will 404 when clicked in 2.0 pages.

File: _includes/install/flow-diagram.md [hunk 7]
```diff
@@ -16,7 +16,7 @@ The diagram shows the following:
 
 	*	For simplicity, get a compressed {{site.data.var.ce}} or {{site.data.var.ee}} [archive]({{ page.baseurl }}/install-gde/prereq/zip_install.html), extract it on your Magento server, and start your installation.
 
-	*	If you are more technical and you are familiar with Composer, get a {{site.data.var.ce}} or {{site.data.var.ee}} [metapackage]({{ page.baseurl }}/install-gde/prereq/integrator_install.html).
+	*	If you are more technical and you are familiar with Composer, get a {{site.data.var.ce}} or {{site.data.var.ee}} [metapackage]({{ page.baseurl }}/install-gde/composer.html).
```
Comment (lorikrell) [inline _includes/install/flow-diagram.md]: Possible 2.0 using include? If used will 404 when clicked in 2.0 pages.

File: _includes/install/releasenotes/ce_install_20.md [hunk 8]
```diff
@@ -10,7 +10,7 @@ See one of the following sections:
 ### Get the Magento Open Source software using Composer {#install-ce-composer}
 {:.no_toc}
 
-This software is available from `repo.magento.com`. Before installing the Open Source software using Composer, familiarize yourself with the Composer metapackage  <a href="{{ page.baseurl }}/install-gde/prereq/integrator_install.html" target="_blank">prerequisites</a>, then run 
+This software is available from `repo.magento.com`. Before installing the Open Source software using Composer, familiarize yourself with the Composer metapackage  <a href="{{ page.baseurl }}/install-gde/composer.html" target="_blank">prerequisites</a>, then run
```
Comment (lorikrell) [inline _includes/install/releasenotes/ce_install_20.md]: Possible 2.0 using include? If used will 404 when clicked in 2.0 pages.

File: _includes/install/releasenotes/ee_install_20.md [hunk 9]
```diff
@@ -10,7 +10,7 @@ See one of the following sections:
 ### Get Magento Commerce using Composer {#install-rc-composer}
 {:.no_toc}
 
-This software is available from `repo.magento.com`. Before installing the Magento Commerce software using Composer,  familiarize yourself with these  <a href="{{ page.baseurl }}/install-gde/prereq/integrator_install.html" target="_blank">prerequisites</a>, then run:
+This software is available from `repo.magento.com`. Before installing the Magento Commerce software using Composer,  familiarize yourself with these  <a href="{{ page.baseurl }}/install-gde/composer.html" target="_blank">prerequisites</a>, then run:
```
Comment (lorikrell) [inline _includes/install/releasenotes/ee_install_20.md]: Possible 2.0 using include? If used will 404 when clicked in 2.0 pages.

File: _includes/php-dev/component-root.md [hunk 10]
```diff
@@ -10,7 +10,7 @@ A component's root directory is the top-level directory for that component under
   * For Admin themes, use `app/design/adminhtml`.
   * For language packages, use `app/i18n`.
 
-* `<Magento install directory>/vendor`: This location is found in the alternative setups where the [`composer create-project`]({{ page.baseurl }}/install-gde/prereq/integrator_install.html) command was used to get a Magento 2 metapackage (which downloads the CE or EE code), or a [compressed Magento 2 archive]({{ page.baseurl }}/install-gde/prereq/zip_install.html) was extracted in order to install Magento. 
+* `<Magento install directory>/vendor`: This location is found in the alternative setups where the [`composer create-project`]({{ page.baseurl }}/install-gde/composer.html) command was used to get a Magento 2 metapackage (which downloads the CE or EE code), or a [compressed Magento 2 archive]({{ page.baseurl }}/install-gde/prereq/zip_install.html) was extracted in order to install Magento.
```
Comment (lorikrell) [inline _includes/php-dev/component-root.md]: Verified this include is used by 2.0.  If used will 404 when clicked in 2.0 pages.

File: _includes/install/flow-diagram.md (hunk 7 above)
Comment (jeff-matthews) [reply _includes/install/flow-diagram.md]: Yes, a search for references to these includes in the 2.0 files did return results. I'll make changes.

File: _includes/install/releasenotes/ce_install_20.md (hunk 8 above)
Comment (jeff-matthews) [reply _includes/install/releasenotes/ce_install_20.md]: Yes, a search for references to these includes in the 2.0 files did return results. I'll make changes.

File: _includes/install/releasenotes/ee_install_20.md (hunk 9 above)
Comment (jeff-matthews) [reply _includes/install/releasenotes/ee_install_20.md]: Yes, a search for references to these includes in the 2.0 files did return results. I'll make changes.

File: _includes/php-dev/component-root.md (hunk 10 above)
Comment (jeff-matthews) [reply _includes/php-dev/component-root.md]: Yes, a search for references to these includes in the 2.0 files did return results. I'll make changes.

File: _includes/install/composer-overview.md (hunk 3 above)
Comment (jeff-matthews) [reply _includes/install/composer-overview.md]: Yes, a search for references to these includes in the 2.0 files did return results. I'll make changes.

File: _includes/php-dev/component-root.md (hunk 10 above)
Comment (lorikrell) [reply _includes/php-dev/component-root.md]: Awesome! Wasn't sure which of these was loading. Thanks, Jeff!

File: _includes/install/file-system-perms-oneuser_22.md (hunk 4 above)
Comment (jeff-matthews) [reply _includes/install/file-system-perms-oneuser_22.md]: I searched the 2.0 files for references to this include, but got zero results.

File: _includes/install/file-system-perms-twouser.md (hunk 5 above)
Comment (jeff-matthews) [reply _includes/install/file-system-perms-twouser.md]: I searched the 2.0 files for references to this include, but got zero results.

File: _includes/install/file-system-perms-twouser_22.md (hunk 6 above)
Comment (jeff-matthews) [reply _includes/install/file-system-perms-twouser_22.md]: I searched the 2.0 files for references to this include, but got zero results.

**Comment by lorikrell** [comment]
Content looks good. Clean and easy to use. Linked content looks good.

**Comment by dshevtsov** [comment]
@jeff-matthews please add `whatsnew`

**Pull request merged**